src/pydash/functions.py

```diff
@@ -50,7 +51,15 @@
 P = ParamSpec("P")
 
 
-class After(t.Generic[P, T]):
+class _WithArgCount(Protocol):
+    func: t.Callable
+
+    @property
+    def _argcount(self) -> t.Optional[int]:
+        return getargcount(self.func, None)
```

Comment:
This was mostly so I can have the nice `_WithArgCount` class and so most of the function wrappers only need to inherit from it and they have the argcount defined.
Otherwise we have to define it everywhere, or in `_WithArgCount`'s `__init__` and call `super().__init__(..)` everywhere.
What do you think about using a `cached_property` otherwise? So it would only be called once and stored in the `__dict__` similar to a `__init__` value and we would still keep the niceness of having a centralized definition of the `_argcount` computation.

Reply:
Only drawback is that I see that the library is supposed to support python3.7 and `cached_property` was introduced in `3.8` so it wouldn't work in `3.7`

Reply:
I went ahead and dropped support for `3.7`. It's past EOL and latest black version dropped it so getting different formatting results between versions, so didn't want to add additional 3.7 workarounds.

So feel free to use the `cached_property`.

Reply:
Just have to rebase with latest `develop` to drop 3.7.

Reply:
rebased and used `cached_property` instead